client/state/comments/reducer.js

```diff
@@ -24,169 +24,165 @@ import {
 	SERIALIZE,
 } from '../action-types';
 import {
-	getCommentParentKey,
-	updateExistingIn
-} from './utils';
-import {
 	PLACEHOLDER_STATE
 } from './constants';
 
 /***
- * Creates a function that updates like count
- * @param {Boolean} like true to like, false to unlike
- * @returns {Function} function that updates like count for comment
- */
-function getLikeSetter( { like = true, likeCount } ) {
-	return ( comment ) => comment.set( 'i_like', like )
-								.update( 'like_count', ( count ) => {
-									if ( typeof likeCount !== 'undefined' ) {
-										return likeCount;
-									}
-
-									return like ? count + 1 : count - 1;
-								} );
-}
-
-/***
- * Updates a specific state in the general state
- * @param {Immutable.Map} state general state for all commentTargetId
- * @param {Object} action action object
- * @param {Number} action.siteId site identifier
- * @param {Number} action.postId post identifier
- * @param {Function|Object} updaterOrValue an updater function or a value
- * @returns {Immutable.Map} new state
- */
-function updateSpecificState( state, action, updaterOrValue ) {
-	const id = getCommentParentKey( action.siteId, action.postId );
-
-	if ( typeof updaterOrValue === 'function' ) {
-		return state.update( id, ( value ) => updaterOrValue( value ) );
-	}
-
-	return state.set( id, updaterOrValue );
-}
-
-/***
  * Comments items reducer, stores a comments items Immutable.List per siteId, postId
- * @param {Immutable.Map} state redux state
+ * @param {Object} state redux state
  * @param {Object} action redux action
- * @returns {Immutable.Map} new redux state
+ * @returns {Object} new redux state
  */
-export function items( state = Immutable.Map(), action ) {
+export function items( state = {}, action ) {
+	const { siteId, postId, commentId, skipSort } = action;
+
 	switch ( action.type ) {
 		case COMMENTS_CHANGE_STATUS:
-			return updateSpecificState( state, action, ( comments = Immutable.List() ) => {
-				updateExistingIn(
-					comments,
-					comment => comment.get( 'ID' ) === action.commentId,
-					comment => comment.set( 'status', action.status )
-				);
-			} );
+			const { status } = action;
+			return {
+				...state,
+				[ `${ siteId }-${ postId }` ]: map( state[ `${ siteId }-${ postId }` ], comment => {
```

Comment:
Is it a problem that we return a new array each time, even if we cannot find the comment to update?